shell/platform/darwin/macos/framework/Source/FlutterTextInputSemanticsObject.h

```diff
@@ -0,0 +1,87 @@
+ public:
+  //---------------------------------------------------------------------------
+  /// @brief      Creates a FlutterTextPlatformNode that uses a
+  ///             FlutterTextField as its NativeViewAccessible.
+  /// @param[in]  delegate          The delegate that provides accessibility
+  ///                               data.
+  /// @param[in]  engine            The engine that is used for querying
+  ///                               the information about FlutterView and
+  ///                               FlutterViewController. The engine must
+  ///                               outlive the lifetime of this ax platform
+  ///                               node.
+  explicit FlutterTextPlatformNode(FlutterPlatformNodeDelegate* delegate, FlutterEngine* engine);
+  ~FlutterTextPlatformNode() override;
+
+  //------------------------------------------------------------------------------
+  /// @brief Gets the bounds of this platform node relative to the NSWindow. This
+  ///        is used by the FlutterTextField to get its frame rect.
+  NSRect GetWindowFrame();
+
+  // |ui::AXPlatformNodeBase|
+  gfx::NativeViewAccessible GetNativeViewAccessible() override;
+
+ private:
+  FlutterEngine* engine_;
+  FlutterTextField* native_text_field_;
+
+  //------------------------------------------------------------------------------
+  /// @brief Ensures the FlutterTextField is attached to the FlutterView. This
+  ///        method returns true if it is attached after this method excuted.
+  ///        If this method returns false, that means the FlutterTextField cannot
+  ///        attach to the FlutterView. This can happen when the FlutterEngine does
+  ///        not have a FlutterViewController or the FlutterView is not loaded yet.
+  bool EnsureAttachedToView();
+
+  //------------------------------------------------------------------------------
+  /// @brief Detaches the FlutterTextField from the FlutterView if it is not
+  ///        already detached.
+  void EnsureDetachedFromView();
+};
+
+}  // namespace flutter
+
+/**
+ * A native text field that represents the NativeViewAccessible for the FlutterTextPlatformNode
+ *
+ * The NSAccessibility protocol does not provide full support for text editing. This native text
+ * field is used to get around this problem. The FlutterTextPlatformNode creates a hidden
+ * FlutterTextField and makes macOS think that it is interacting with a real native text field.
+ * This causes VoiceOver to announce text editing feedbacks.
+ *
+ * All of the text editing events in this native text field are redirected to the
+ * FlutterTextInputPlugin and will eventually update this text field through text input channel.
+ */
+@interface FlutterTextField : NSTextField
+
+/**
+ * Initializes a FlutterTextField.
+ */
+- (instancetype)initWithPlatformNode:(flutter::FlutterTextPlatformNode*)node
+                     withFieldEditor:(FlutterTextInputPlugin*)plugin;
```

Comment:
As far as I can tell it is a programming error to allow an instance of this class to outlive `plugin`.
- That's extremely important information to document here, and
- You cannot actually enforce this kind of restriction in the general case; there are plenty of cases where the framework may end up with a temporary reference to an object longer than you wanted it to be alive. Assuming you can manage the exact lifetime of a reference-counted object is almost always a recipe for subtle crashes later; you should be providing an explicit way to nil out the editor which whatever is managing relative (primary) lifetimes can call at the appropriate moment.